zen/render-2d.c

```diff
@@ -9,21 +9,83 @@
 #include "zen/scene/screen.h"
 #include "zen/scene/view.h"
 
+typedef void (*zn_surface_iterator_func_t)(struct wlr_surface *surface,
+    struct wlr_renderer *renderer, const struct wlr_fbox *box);
+
+struct zn_surface_iterator_data {
+  zn_surface_iterator_func_t iterator;
+
+  struct zn_view *view;
+  struct wlr_renderer *renderer;
+};
+
 static void
-zn_render_2d_view(struct zn_view *view, struct wlr_renderer *renderer)
+zn_render_get_surface_fbox(struct wlr_surface *surface, struct zn_view *view,
+    int surface_x, int surface_y, struct wlr_fbox *surface_box)
 {
-  struct wlr_surface *wlr_surface = view->impl->get_wlr_surface(view);
-  struct wlr_texture *texture = wlr_surface_get_texture(wlr_surface);
-  struct wlr_fbox fbox;
+  surface_box->x = view->x + surface_x;
+  surface_box->y = view->y + surface_y;
+  surface_box->width = surface->current.width;
+  surface_box->height = surface->current.height;
+}
+
+static void
+zn_render_wlr_surface_iterator(
+    struct wlr_surface *surface, int surface_x, int surface_y, void *user_data)
+{
+  struct zn_surface_iterator_data *data = user_data;
+  struct wlr_fbox surface_box;
+
+  zn_render_get_surface_fbox(
+      surface, data->view, surface_x, surface_y, &surface_box);
+
+  data->iterator(surface, data->renderer, &surface_box);
+}
 
+static void
+zn_render_surface_iterator(struct wlr_surface *surface,
```

Comment:
The function name seems arbitrary.

```suggestion
zn_render_2d_render_surface_iterator(struct wlr_surface *surface,
```